src/Popover/Popover.tsx

```diff
@@ -377,45 +448,45 @@ export default class Popover extends React.Component {
   };
 
   getVisiblePopover = () => {
-    const { popover } = this.state;
+    const popover = this.state.popover!; // TODO(gabro): is this safe?
     const popoverProps = this.getPopoverProps();
 
     if (popoverProps.auto) {
       // give priority to the position passed by the user as _.uniq should maintain the order
-      const positions = uniq([popoverProps.position, 'top', 'bottom', 'left', 'right']);
-
-      const popoverStyle = positions.reduce((acc, p) => {
-        if (acc === null) {
-          // give priority to the couple position/anchor passed by the user as _.uniq should maintain the order
-          const anchors = uniq(
-            (p === popoverProps.position ? [popoverProps.anchor] : []).concat(['center', 'start', 'end'])
-          );
-
-          return anchors.reduce((workingPopoverStyle, a) => {
-            if (workingPopoverStyle === null) {
-              const _popoverStyle = this.computePopoverStyle(p, a);
-
-              // not enough space
-              if (
-                _popoverStyle.top < 0 || _popoverStyle.top + popover.height > window.innerHeight ||
-                _popoverStyle.left < 0 || _popoverStyle.left + popover.width > window.innerWidth
-              ) {
-                return null;
+      const positions = uniq<PopoverProps.Position>([popoverProps.position, 'top', 'bottom', 'left', 'right']);
```

Comment:
for some reason `uniq` didn't infer the type correctly.

Reply:
It's probably the same old story with string literal unions: `[popoverProps.position, 'top', 'bottom', 'left', 'right']` is inferred as `(Position | string)[]`